utils/run_test.sh

```diff
@@ -53,18 +36,20 @@ run_test()
     export D_LOG_FILE="/tmp/daos_${b}-${log_num}.log"
     echo "Running $* with log file: ${D_LOG_FILE}"
 
-    # We use flock as a way of locking /mnt/daos so multiple runs can't hit it
-    #     at the same time.
+    export TNAME="${b}-${log_num}"
+
     # We use grep to filter out any potential "SUCCESS! NO TEST FAILURES"
     #    messages as daos_post_build.sh will look for this and mark the tests
     #    as passed, which we don't want as we need to check all of the tests
     #    before deciding this. Also, we intentionally leave off the last 'S'
     #    in that error message so that we don't guarantee printing that in
     #    every run's output, thereby making all tests here always pass.
-    if ! time $lock_test "$@"; then
-        echo "Test $* failed with exit status ${PIPESTATUS[0]}."
-        ((failed = failed + 1))
-        failures+=("$*")
+    time eval "${VALGRIND_CMD}" "$@"
+    retcode=$?
+    if [ "${retcode}" -ne 0 ]; then
```

Comment:
Test for failure directly instead of looking at exit codes:

```suggestion
    if ! time eval "${VALGRIND_CMD}" "$@"; then
      retcode=${PIPESTATUS[0]}
```